### PR TITLE
Upgrade tox and linter to conventions used by other projects

### PR DESCRIPTION
**Pull request by f4str**
Updated `tox` to use `flake8` for the linter and changed `black` and `isort` configurations to match linting and styling conventions used by all other projects for consistency (most noticeably changing line-length from 88 to 100).

`mypy`'s newest release allows configurations to be stored in `pyproject.toml` instead of requiring its own `mypy.ini`.

Cleaned up the `setup.py` dev requirements as they are no longer needed since `tox` will automatically install them in a `virtualenv` wrapper when needed. 

All changes made are purely linting fixes (most of which are by `black` and `isort` auto-formatting) such as:

- removing unused imports
- removing unused variables
- docstring capitalization/formatting
- changing type annotations
- consolidating/expanding code into less/more lines

### CODE REVIEW

File: server/pyproject.toml [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
-exclude='\.eggs|\.git|\.mypy_cache|\.tox|\.venv|_build|buck-out|build|dist|venv'
+target-version = ["py38"]
+exclude = '\.eggs|\.git|\.mypy_cache|\.tox|\.env|\.venv|env|venv|_build|buck-out|build|dist'
 
 [tool.isort]
-line_length = 88
-use_parentheses = true
-include_trailing_comma = true
-multi_line_output = 3
+profile = "black"
```
Comment (subdavis) [inline server/pyproject.toml:8]: This was like the greatest addition to isort ever.

File: server/tests/test_deserialize_viame_csv.py [hunk 2]
```diff
@@ -8,20 +8,22 @@
...
         [
-            # all frames in a track must be of the same type, so the type name for 0 will be ignored
+            # all frames in a track must be the same type, so the type name for 0 will be ignored
             "0,1.png,0,884.66,510,1219.66,737.66,1,-1,ignored,0.98",
             "0,2.png,1,111,222,3333,444,1,-1,typestring,0.55",
             "1,1.png,0,747,457,1039,633,1,-1,type2,1",
             # Keypoint testing with HeadTails Line
-            "2,3.png,2,10,50,20,35,1,-1,type3,0.765,(kp) head 22.4534 45.6564,(kp) tail 55.232 22.3445",
+            "2,3.png,2,10,50,20,35,1,-1,type3,0.765,(kp) head 22.4534 45.6564,"
+            "(kp) tail 55.232 22.3445",
```
Comment (subdavis) [inline server/tests/test_deserialize_viame_csv.py:17]: I don't think this is OK style.  Could you either come up with a way to let these string literals span their full width?  I don't even care if you have to disable this rule for this file, this makes the code harder to read IMO.

File: server/tests/test_deserialize_viame_csv.py [hunk 3]
```diff
@@ -271,16 +273,24 @@
...
-            "1,3.png,2,747,457,1039,633,1,-1,typestring,1,(trk-atr) booleanAttr true,(atr) DetectionPredefinedValue value3",
-            "1,3.png,3,884.66,510,1219.66,737.66,1,-1,typestring,1,(trk-atr) booleanAttr true,(atr) DetectionPredefinedValue value1",
+            "1,1.png,0,884.66,510,1219.66,737.66,1,-1,typestring,1,(atr) DetectionNumber 2.002,"
+            "(atr) DetectionPredefinedValue value1",
+            "1,2.png,1,111,222,3333,444,1,-1,typestring,1,(trk-atr) booleanAttr true,"
+            "(atr) DetectionPredefinedValue value2",
+            "1,3.png,2,747,457,1039,633,1,-1,typestring,1,(trk-atr) booleanAttr true,"
+            "(atr) DetectionPredefinedValue value3",
+            "1,3.png,3,884.66,510,1219.66,737.66,1,-1,typestring,1,(trk-atr) booleanAttr true,"
+            "(atr) DetectionPredefinedValue value1",
```
Comment (subdavis) [inline server/tests/test_deserialize_viame_csv.py:293]: same

File: server/tests/test_deserialize_viame_csv.py [hunk 4]
```diff
@@ -8,20 +8,22 @@
...
             # Keypoint without HeadTails Line
             "2,4.png,3,10,50,20,35,1,-1,type3,0.765,(kp) head 22.4534 45.6564",
             # Multiple ConfidencePair Test
             "3,4.png,4,10,10,20,20,1,-1,type1,0.89,type2,0.65",
             # Polygon Test with floats
             "4,5.png,5,10,10,20,20,1,-1,type1,0.89,type2,0.65,(poly) 1 2.34 3 4 5 6 7 8.08 9 10",
             # Track and Frame Attr testing
-            "5,6.png,6,10,10,20,20,1,-1,type1,0.89,(atr) attrNAME spaced attr name,(trk-atr) booleanAttr true",
+            "5,6.png,6,10,10,20,20,1,-1,type1,0.89,(atr) attrNAME spaced attr name,"
+            "(trk-atr) booleanAttr true",
```
Comment (subdavis) [inline server/tests/test_deserialize_viame_csv.py:26]: same

File: server/tests/test_serialize_viame_csv.py [hunk 5]
```diff
@@ -201,8 +197,10 @@
             },
         },
         [
-            "0,1.png,0,884,510,1219,737,1.0,-1,typestring,1.0,(atr) detectionAttr frame 0 attr,(trk-atr) trackATTR TestTrack ATTR With Space",
-            "0,2.png,1,111,222,3333,444,1.0,-1,typestring,1.0,(atr) detectionAttr frame 1 attr,(trk-atr) trackATTR TestTrack ATTR With Space",
+            "0,1.png,0,884,510,1219,737,1.0,-1,typestring,1.0,(atr) detectionAttr frame 0 attr,"
+            "(trk-atr) trackATTR TestTrack ATTR With Space",
+            "0,2.png,1,111,222,3333,444,1.0,-1,typestring,1.0,(atr) detectionAttr frame 1 attr,"
+            "(trk-atr) trackATTR TestTrack ATTR With Space",
```
Comment (subdavis) [inline server/tests/test_serialize_viame_csv.py:203]: same

**Review by subdavis — CHANGES_REQUESTED** [review]
Please fix string literal line wraps.  Everything else looks good!

File: server/tests/test_deserialize_viame_csv.py (hunk 2 above)
Comment (BryonLewis) [reply server/tests/test_deserialize_viame_csv.py:17]: should the content for these tests be moved to `./testutils` and loaded in like for the image-sort.  Then you no longer are worrying about linting and formatting test data?

Comment (subdavis) [reply server/tests/test_deserialize_viame_csv.py:17]: They're paired input/expected, so you'd have to organize them as such. 

I don't feel that strongly about it, but even if you moved these, you still couldn't have str literals longer than 100chars, which as a concept I would like to resolve.

**Review by subdavis — APPROVED** [review]

**Comment by f4str** [comment]
Fixed the string wrapping and added the `flake8` config to ignore line length linting for `tests/`

Also added the `tox -e docs` option to run `mkdocs` and serve the site. Thus `mkdocs` was removed from the dev requirements since `tox` will automatically install it. This is documented in `server/README.md`

Finally changed the version number in `server/setup.py` to match `client/package.json`.

**Pull request merged**